Documentation/ApiOverview/Fluid/_CustomViewHelper/_GravatarTagBasedViewHelper.php

```diff
@@ -13,14 +13,21 @@ final class GravatarViewHelper extends AbstractTagBasedViewHelper
     public function initializeArguments(): void
     {
         parent::initializeArguments();
-        $this->registerArgument('emailAddress', 'string', 'The email address to resolve the gravatar for', true);
+        $this->registerArgument(
+            'emailAddress',
+            'string',
+            'The email address to resolve the gravatar for',
+            true,
+        );
+        // The alt argument will be automatically registered
```

Comment:
Just checked the blog_example tests. There is no "alt" attribute: https://github.com/TYPO3-Documentation/blog_example/blob/main/Tests/Functional/ViewHelpers/GravatarViewHelperTest.php#L49

Reply:
I found this commit: https://github.com/TYPO3/Fluid/commit/68176ee7e628fca2867eaa43c7a099c0e1106f5f
In that case "alt" has to be defined before which it is not. It's not in the list of additionalAttributes and has no default. I'm very interessted in that feature, can you give me a link to a changelog, code-line or anything else? Thank you for your work at fluid code...just see all your last commits there. Wohouu

Reply:
Thanks! :) 
Core RST: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/13.2/Feature-104223-UpdateFluidStandaloneTo212.html#feature-104223-update-fluid-standalone-to-2-12
Fluid PR: https://github.com/TYPO3/Fluid/pull/859

Reply:
I don't get it. I have current GIT clone of TYPO3 core. It's still the job of the ViewHelper to decide, if an attribute will be added or not:

https://github.com/TYPO3/typo3/blob/main/typo3/sysext/fluid/Classes/ViewHelpers/ImageViewHelper.php#L233-L236

So, as long as Gravatar ViewHelper does not handle the "alt" attribute, it will not be rendered.

Maybe we should move our discussion to slack, if you want. You find me with user froemken. Have a nice evening

Reply:
Resolved too early. Sorry. As you have added the "alt" attribute manually now, can we remove that line now?

Reply:
No teally sure what causes your confusion. You **do not have to add the alt attribute manually** in a TagBasedViewHelper and I did not do so.

Reply:
THX for clarify at Slack. I really don't get it with all the ChangeLogs.

Reply:
Just for clarification here: This feature was developed for Fluid 4, then backported and released with Fluid 2.12. Then it was applied to TYPO3-main with the RST file linked above and as `composer update ...` to 12.4.

It is available to all TYPO3 instances that use Fluid >= 2.12, which might even be TYPO3 v11 if the composer dependency updates have been performed.